copybookreader-plugins/docs/CopyBookReader-batchsource.md

```diff
@@ -0,0 +1,99 @@
+# CopybookReader plugin
+
+
+Description
+-----------
+This is a source plugin that would allow users to read and process mainframe files defined using COBOL Copybook.
+This is the basic first implementation.
+
+Use Case
+--------
+
+It's basically used for reading flat file or dataset that is generated on a z/OS IBM mainframe based on a fixed length
+COBOL Copybook. This will also work on AS/400 computers.
+So, if a customer has flat files on HDFS that can be parsed using simple COBOL Copybook, then applying the Copybook
+one is able to read the file and it's fields easily.
+
+
+Properties
+----------
+
+**CopybookContents :** Contents of the COBOL Copybook file which will contain the data structure
+User will have to paste the entire contents of the Copybook file in the test area.
+The Copybook contents should be in sync with the underlying binary data to be read.
+First implementation does not handle complex nested structures of COBOL Copybook
+Also it does not handle Redefines or iterators in the structure
+
+**binaryFilePath   :** Complete path of the .bin to be read.This will be a fixed length binary format file,that
+matches the Copybook.
+Supports compressed files - Native Compressed Codec
+
+**fileStructure    :** Copybook file structure. For the current implementation only fixed length flat files
+will be read.
+fixed length binary format is supported in the first implementation
+
+**schema:** The schema for the data as it will be formatted in CDAP.
+
+
+Example
+-------
+
+This example reads data from a local binary file "file:///home/cdap/cdap/DTAR020_FB.bin"  and parses it using the schema given in the text area "COCOL CopyBook contents"
+It will generate structured records with either the output schema (if specified by the user) or with the default schema as is specified in the text area.
```

Comment:
If this means that the user species the output schema in the text area, perhaps this should be:
```
It will generate structured records using either the default schema 
or with the output schema specified by the user in the text area.